### DISCUSSION TITLE
Is there a way to limit Kafka reads until an offset

### DISCUSSION THREAD

**Discussion by sxm51** [ClickHouse/ClickHouse#54612] [Q&A] [answered]
Hi,

Is there a way to limit Kafka read for a consumer-group/topic until an offset.

Context:
Have a case where I want to fill lot of historical data into a table from a Kafka topic, and at the same time start updating the table with new data as it comes.
The goal is to let table serve the new data, and have the (slow) background process that backfill the data at the same time.
Implementation wise -  thinking we'll have two Kafka tables and MV, each Kafka table with a different consumer group.
Is it possible to implement it using the existing ClickHouse Kafka integration.

**Comment by UnamedRus** [comment, marked as answer]
You can add WHERE condition to historical MV, like
WHERE (_partition = xxxx AND  _offset = xxxx) OR (_partition = ...)

    _topic AS topic,
    _partition AS partition,
    _offset AS offset,
    _raw_message AS raw,
    _error AS error

**Reply by sxm51** [reply]
Great.
Didn't know we had `_offset` virtual column.
